bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -83,6 +85,36 @@ impl Room {
     pub(crate) fn with_timeline(inner: SdkRoom, timeline: TimelineLock) -> Self {
         Room { inner, timeline }
     }
+
+    fn message_filtered_timeline_builder(
+        &self,
+        internal_id_prefix: Option<String>,
+        allowed_message_types: Vec<RoomMessageEventMessageType>,
+        date_divider_mode: DateDividerMode,
+    ) -> TimelineBuilder {
```

Comment:
It's a bit weird to have a function returning a builder, but with optional arguments. Those optional arguments could become setters on the builder itself, instead, and could be exposed in the public API. This way, we could allow for more control over how the timelines are created, with a likely simpler API. What you made here works, but will need to change the next time anyone requires a new configuration. What do you think?

Reply:
I'm a tad confused, don't think I understand what you mean.

Why is it weird? It's just a private helper method to reuse configuration code between the 2 different message filtered timelines🤔 

The only optional argument here is `with_internal_id_prefix` which is already part of both the builder and the existing timeline creation methods. All `message_filtered_timeline_builder ` is make is so it doesn't need to be unwrapped in 2 different place.

Either way, this PR is not about refactoring how timelines are created. If you wish we can tackle that separately.